### PR TITLE
gvisor: `20231113.0` -> `20240311.0-unstable-2024-03-25`

### PR DESCRIPTION
**Pull request by msanft**
## Description of changes

Update gVisor to [`e3cf008`](https://github.com/google/gvisor/commit/e3cf008ab18671c8d1f58c9a04cb971573bcb724)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by msanft** [comment]
@ofborg test gvisor

File: pkgs/applications/virtualization/gvisor/default.nix [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 buildGoModule {
   pname = "gvisor";
-  version = "20231113.0";
+  version = "20240216.0";
```
Comment (katexochen) [inline pkgs/applications/virtualization/gvisor/default.nix:13]: Not sure I get this right but the commit says: `Merge release-20240212.0-16-gaa1a66353`,
so I would assume version should be `20240212.0` or `20240212.0-16`?

**Review by katexochen — CHANGES_REQUESTED** [review]

**Comment by nevivurn** [comment]
That commit message is generated by [`tools/go_branch.sh`](https://github.com/google/gvisor/blob/master/tools/go_branch.sh) and  is the output from `git describe --always`. As-is, the latest release is `20240212.0` and the currently pinned commit in this PR google/gvisor@e3cf008ab18671c8d1f58c9a04cb971573bcb724 ought to be referred to as version `20240212.0-unstable-2024-02-16`.

Unfortunately there is no one commit on the `go` branch that corresponds to `release-20240212.0` exactly, the closest commit is google/gvisor@1796ac7f1 which includes the changes from google/gvisor@c2bcdb73a also.

I didn't realize this in #268853, I should have either pinned google/gvisor@8a880e3b2 (the correct commit for `20231113.0`) or named the version `20231120.0-unstable-2023-11-20`.

**Comment by katexochen** [comment]
@ofborg test gvisor

**Review by nevivurn — APPROVED** [review]

**Review by katexochen — APPROVED** [review]

**Pull request merged**